engine/src/main/java/org/terasology/engine/world/chunks/pipeline/ChunkProcessingInfo.java

```diff
@@ -11,30 +11,25 @@
 
 import java.util.List;
 import java.util.concurrent.Future;
+import java.util.concurrent.locks.ReentrantLock;
 
 public final class ChunkProcessingInfo {
+    public final ReentrantLock lock = new ReentrantLock();
```

Comment:
It's used by the ChunkProcessingPipeline worker threads so that they're not working on the same chunk at the same time. I should probably add a javadoc comment for that.